### PR TITLE
Rework the feature selection

### PR DESCRIPTION
**Pull request by coolo**
- Do not monkey patch Feature class, but use our custom config
  loading class to merge beta features if configured
- Provide a beta switching function in Feature and use it from
  User.session=

### CODE REVIEW

**Comment by coolo** [comment]
This is the same approach but possibly more acceptable - I left the cleanup of the defaults in config/features.yml for a later moment and made the feature switch when setting the user session. This leaks less, but makes the Feature gem a clear dependency of the user model.

And @vpereira before you ask: the unit test only tests the Feature class as used by user, not the integration into the User model. That is at the moment only implicitly tested in controller tests e.g. for kiwi_image that verify you get a 404 if you don't have the feature enabled.

**Comment by vpereira** [comment]
> And @vpereira before you ask: the unit test only tests the Feature class as used by user, not the integration into the User model. That is at the moment only implicitly tested in controller tests e.g. for kiwi_image that verify you get a 404 if you don't have the feature enabled.

But does it make sense to couple the ```User.session=``` with ```Feature``` if an user just use the API?

File: src/api/app/models/user.rb [hunk 1]
```diff
@@ -244,9 +244,10 @@ def self.admin_session?
     current && current.is_admin?
   end
 
-  # set the user as current session user (should be real user)
+  # set the user as current session user (should be real user or nil)
   def self.session=(user)
     Thread.current[:user] = user
+    Feature.use_beta_features(user.try(:in_beta) || false)
```
Comment (vpereira) [inline src/api/app/models/user.rb]: We use ```User.session``` everywhere, specially models, model specs and API. I don't think this WebUI specific setting should be necessary for them..

Comment (coolo) [reply src/api/app/models/user.rb]: what makes you think features are webui specific? It's just a question of time til (staging) API enables some features only for some users.

But even if not: The use_beta_features function only resets some variables. And User.session= is the right place IMO

**Comment by hennevogel** [comment]
handled in #7880 now... 

```


                               _                        
                            .-" l                       
                          .'   /                        
           _.------------'    :                         
      .-"""                    "-.                      
    .'                            "-.                   
 _.-'    __                          "-.                
(---"""  "        _                     "-.             
 """"""""-----.-"" `.                      `.           
               \ `.__;________               `.         
                `.:           """"---...       `.       
                  `                     """--..  \--""-.
                                              :   `.   :
                                               \    .-" 
                                                "-._l   
```

**Pull request closed without merging**